components/UI/MultiSelect.tsx

```diff
@@ -0,0 +1,185 @@
+  }
+
+
+  useEffect(() => {
+    setIsDropdownOpen(false)
+  }, [parentClick])
+
+  useEffect(() => {
+    onChange(selectedOptions)
+  }, [selectedOptions, onChange])
+
+  useEffect(() => {
+    if (grow && options) {
+      debugger
+      let seedLength: number
+      const longestLabel = options.reduce((longest, option) => {
+        return option.label.length > longest.length ? option.label : longest
+      }, '')
+      seedLength = longestLabel.length > title.length ? longestLabel.length : title.length
+
+      const averageFontWidth = 0.6 * parsedFontSize // Assuming an average font width of 7 pixels (adjust as needed)
+      const extraSpace = 5 // Additional space for left and right side padding
+      const font = `${fontSize} Arial`
+      //const calculatedWidth = seedLength * averageFontWidth + extraSpace
+      let calculatedWidth = getStringWidth(longestLabel, font);
+      setCalculatedWidth(calculatedWidth)
+    }
+  }, [options, grow])
+
+  return (
+    <div id="MultiSelect Level 0 Div" style={{ backgroundColor:'blue',position: 'relative', width: grow ? calculatedWidth : width }}>
+      <div id="MultiSelect Level 1  Div">
+        <div
+          id="MultiSelect div button"
+          onMouseEnter={toggleDropdown}
+          style={{
+            backgroundColor: '#EEE',
+            cursor: 'pointer',
+            padding: '1px',
+            border: '1px solid #CCC',
+            borderRadius: '4px',
+            height: toggleHeight,
+            textAlign: 'center',
+            fontSize: `${fontSize}`,
+          }}
+        >
+          {title}
+        </div>
+        {isDropdownOpen && (
+          <div id="List Div"
+          onMouseEnter={toggleDropdownList}
+          onMouseLeave={toggleDropdownList}
+            style={{
+              position: 'absolute',
+              top: toggleHeight,
+              left: 0,
+              backgroundColor: '#FFF',
+              padding: '10px',
+              border: '1px solid #CCC',
+              borderRadius: '4px',
+              zIndex: 1,
+              maxHeight: '300px', // Adjust the max height as needed
+              overflow: 'auto',
+            }}
+          >
+            {options.map(option => (
+              <MultiSelectOption
+                key={option.value}
+                option={option}
+                fontSize={fontSize}
+                onChange={() => handleOptionClick(option)}
+                isSelected={selectedOptions.some(
+                  selectedOption => selectedOption.value === option.value
+                )}
+              />
+            ))}
+          </div>
+        )}
+      </div>
+      <p style={{ textAlign: 'left' }}>
+        Recipients: {selectedOptions.map(option => option.label).join(', ')}
+      </p>
+    </div>
+  )
+}
+
+interface MultiSelectOptionProps {
+  option: Option
+  onChange: () => void
+  isSelected: boolean
+  fontSize: string
+}
+
+const MultiSelectOption: React.FC<MultiSelectOptionProps> = ({
```

Comment:
This should be in it's own `multi-select-option.tsx` file. That way your code will be easier to read when you want to find something.